src/main/java/com/alipay/oceanbase/rpc/bolt/transport/ObTableRemoting.java

```diff
@@ -132,7 +121,17 @@ public ObPayload invokeSync(final ObTableConnection conn, final ObPayload reques
             // decode ResultCode for response packet
             ObRpcResultCode resultCode = new ObRpcResultCode();
             resultCode.decode(buf);
-
+            // If response indicates the request is routed to wrong server, we should refresh the routing meta.
+            if (response.getHeader().isRoutingWrong()) {
+                String errMessage = TraceUtil.formatTraceMessage(conn, request,
+                        "routed to the wrong server: " + response.getMessage());
+                logger.warn(errMessage);
+                if (needFetchAll(resultCode.getRcode())) {
+                    throw new ObTableRoutingWrongException(errMessage);
```

Comment:
use another new exception?   this exception has been used before and cannot represent that needs to refresh all table entry

Reply:
Fixed